Bond/Core/Buffer.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  The MIT License (MIT)
+//
+//  Copyright (c) 2015 Srdan Rasic (@srdanrasic)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// A simple collection that buffers latest `size` values.
+public struct Buffer<EventType> {
+  
+  /// Internal buffer
+  public var buffer: [EventType] = []
+  
+  /// Buffer size
+  public var size: Int
+  
+  public init(size: Int) {
+    self.size = size
+  }
+  
+  /// Adds the given element to the buffer. If the buffer is already 
+  /// at its capacity, it will discard the oldest element.
+  public mutating func push(event: EventType) {
+    buffer.append(event)
+    if buffer.count > size {
+      buffer.removeAtIndex(0)
```

Comment:
Complexity is O(n).
Maybe it is worth to measure and in case of issues replace it with a simple linked-list? (how much I'd like an std::deque here :)